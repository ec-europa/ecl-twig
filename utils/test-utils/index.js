const Twig = require('twig');
const { format } = require('prettier');

const renderTwigFile = (file, options, cb) =>
  Twig.renderFile(file, options, (err, html) =>
    cb(err, format(html, { parser: 'html' }))
  );

const renderTwigFileAsNode = (file, options) =>
  new Promise((resolve, reject) =>
    Twig.renderFile(file, options, (err, html) => {
      if (err) return reject(err);

      const div = document.createElement('div');
      div.innerHTML = html.trim();

      return resolve(div.firstChild);
    })
  );

module.exports = {
  renderTwigFile,
  renderTwigFileAsNode,
};
