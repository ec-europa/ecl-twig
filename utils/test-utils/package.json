{
  "name": "@ecl-twig/test-utils",
  "version": "2.29.0",
  "description": "A local test helper",
  "main": "index.js",
  "private": true,
  "devDependencies": {
    "deepmerge": "4.2.2"
  }
}
