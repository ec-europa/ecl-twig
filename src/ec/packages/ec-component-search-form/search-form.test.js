import path from 'path';
import { renderTwigFileAsNode } from '@ecl-twig/test-utils';

describe('EC - Search Form', () => {
  const template = path.resolve(__dirname, './search-form.html.twig');
  const render = params => renderTwigFileAsNode(template, params);
  const defaultIconPath = 'static/icons.svg';

  describe('Default', () => {
    test('renders correctly', () => {
      expect.assertions(1);

      const defaultDataStructure = {
        text_input: {
          id: 'input-email',
          name: 'email',
          extra_classes: 'ecl-search-form__text-input',
        },
        button: {
          variant: 'search',
          icon: {
            type: 'general',
            name: 'search',
            path: defaultIconPath,
            size: 'fluid',
          },
          label: 'Search',
          extra_classes: 'ecl-search-form__button',
        },
      };

      return expect(render(defaultDataStructure)).resolves.toMatchSnapshot();
    });
  });
});
