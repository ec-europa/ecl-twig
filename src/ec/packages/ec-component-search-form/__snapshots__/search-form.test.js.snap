// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EC - Search Form Default renders correctly 1`] = `
<form
  class="ecl-search-form"
  role="search"
>
  <div
    class="ecl-form-group ecl-form-group--text-input"
  >
    <input
      class="ecl-text-input ecl-search-form__text-input"
      id="input-email"
      name="email"
      placeholder=""
      type="text"
    />
  </div>
  <button
    class="ecl-button ecl-button--search ecl-search-form__button"
    type="submit"
  >
    <span
      class="ecl-button__container"
    >
      <span
        class="ecl-button__label"
        data-ecl-label=""
      >
        Search
      </span>
      <svg
        aria-hidden="true"
        class="ecl-icon ecl-icon--fluid ecl-button__icon ecl-button__icon--after"
        focusable="false"
      >
        <use
          xlink:href="static/icons.svg#general--search"
        />
      </svg>
    </span>
  </button>
</form>
`;
